catalog/sql/sql.go

```diff
@@ -524,9 +525,14 @@ func (c *Catalog) DropNamespace(ctx context.Context, namespace table.Identifier)
 		return fmt.Errorf("%w: %s", catalog.ErrNoSuchNamespace, nsToDelete)
 	}
 
-	tbls, err := c.ListTables(ctx, namespace)
-	if err != nil {
-		return err
+	tbls := make([]table.Identifier, 0)
+	iter := c.ListTables(ctx, namespace)
+
+	for tbl, err := range iter {
```

Comment:
No need to gather all the tables, as so as you get 1 table you can break and return the error below

Reply:
Nice catch. Thank you for pointing it out :pray: